src/test/scala-2.11/markets/actors/participants/OrderTrackerSpec.scala

```diff
@@ -0,0 +1,98 @@
+
+import markets.MarketsTestKit
+import markets.tickers.Tick
+import markets.actors.{Accepted, Filled}
+import markets.orders.limit.LimitAskOrder
+import markets.orders.market.MarketBidOrder
+import markets.actors.participants.strategies.ConstantOrderIssuingStrategy
+import markets.orders.{AskOrder, BidOrder}
+import markets.tradables.Tradable
+import org.scalatest.{FeatureSpecLike, GivenWhenThen, Matchers}
+
+import scala.util.Random
+
+
+/** Test specification for any actor mixing in the `OrderTracker` trait. */
+class OrderTrackerSpec extends TestKit(ActorSystem("OrderTrackerSpec"))
+  with MarketsTestKit
+  with FeatureSpecLike
+  with GivenWhenThen
+  with Matchers {
+
+  /** Shutdown TestSystem after running tests. */
+  def afterAll(): Unit = {
+    system.terminate()
+  }
+
+  val prng = new Random(42)
+
+  val tradable = Tradable("GOOG")
+
+  val market = TestProbe()
+
+  val initialTick = Tick(1, 1, 1, 1, timestamp())
+
+  val ticker = Agent(initialTick)(system.dispatcher)
+
+  feature("An OrderTracker should be able to add and remove outstanding orders.") {
+
+    val askOrderIssuingStrategy = ConstantOrderIssuingStrategy[AskOrder](Some(2), 1, Some(tradable))
+    val bidOrderIssuingStrategy = ConstantOrderIssuingStrategy[BidOrder](Some(1), 1, Some(tradable))
+    val props = TestOrderTracker.props(askOrderIssuingStrategy, bidOrderIssuingStrategy)
+    val orderTrackerRef = TestActorRef[TestOrderTracker](props)
+    val orderTrackerActor = orderTrackerRef.underlyingActor
+
+    orderTrackerRef ! Add(tradable, market.ref, ticker)
+
+    scenario("An OrderTracker receives a Filled message...") {
+
+      Given("An OrderTracker with outstanding orders...")
+      val price = randomLimitPrice(prng)
+      val quantity1 = randomQuantity(prng)
+      val order1 = LimitAskOrder(orderTrackerRef, price, quantity1, timestamp(), tradable, uuid())
+      val quantity2 = randomQuantity(prng)
+      val order2 = MarketBidOrder(orderTrackerRef, quantity2, timestamp(), tradable, uuid())
+      orderTrackerRef tell(Accepted(order1), testActor)
+      orderTrackerRef tell(Accepted(order2), testActor)
+
+      When("An OrderTracker receives a Filled message with no residual order...")
+      val filled = Filled(order1, None)
+      orderTrackerRef tell(filled, testActor)
+
+      Then("...it should remove the filled order.")
+      orderTrackerActor.outstandingOrders.headOption should be(Some(order2))
+
+      When("An OrderTracker receives a Filled message with some residual order...")
+      val residualQuantity = randomQuantity(prng, upper=quantity2)
+      val(_, residualOrder) = order2.split(residualQuantity)
+      val partialFilled = Filled(order2, Some(residualOrder))
+      orderTrackerRef ! partialFilled
+
+      Then("...it should remove the filled order and replace it with the residual order.")
+      orderTrackerActor.outstandingOrders.headOption should be(Some(residualOrder))
+
+    }
+
+  }
+  
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Whitespace at end of line](https://www.codacy.com/app/davidrpugh/markets-sandbox/file/2586596028/issues/source?bid=2921361&fileBranchId=3304514#l97)